tests/rptest/services/redpanda.py

```diff
@@ -834,7 +847,72 @@ def __init__(self,
         self.listener_authn_method = listener_authn_method
 
 
-class RedpandaServiceBase(Service):
+class RedpandaServiceABC(ABC):
+    """A base class for all Redpanda services. This lowest-common denominator
+    class has both implementation and abstract methods, only for methods which
+    can be implemented by all services. Any methods which the service should
+    implement should be @abstractmethod in order to ensure they are implemented
+    by base classes.
+
+    If a method can be implemented by more than one service implementation, but
+    not all of them, it does NOT belong here.
+    """
+
+    context: TestContext
+    logger: Logger
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Now ensure that this base is part of the right type of object.
+        # For these checks to work, this __init__ method should appear in
+        # the MRO after all the __init__ methods that would set these properties
+        self._check_attr('context', TestContext)
+        self._check_attr('logger', Logger)
+
+    def healthy(self) -> bool:
```

Comment:
Good question, it's because cloud service does not implement this yet (we have a bug for it). In any case, we don't need this in the ABC as we don't use it elsewhere in the ABC, so I have removed it.